### ISSUE TITLE
backend_setup role fails with ansible-core 2.13.3 and Python 3.9

### ISSUE THREAD

**Issue by anoopcs9** [gluster/gluster-ansible-infra#135]
We have been noticing the following failure since the update to ansible 2.13.3 with its dependency on Python 3.9 on CentOS Stream 8:
```
TASK [gluster.infra/roles/backend_setup : Group devices by volume group name, including existing devices] ***
fatal: [storage0]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was:
'str object' has no attribute 'vgname'\n\nThe error appears to be in '/etc/ansible/roles/gluster.infra/roles/
backend_setup/task/get_vg_groupings.yml': line 3, column 3, but may\nbe elsewhere in the file depending on
the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Group devices by volume group
name, including existing devices\n  ^ here\n"}
fatal: [storage1]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was:
'str object' has no attribute 'vgname'\n\nThe error appears to be in '/etc/ansible/roles/gluster.infra/roles/
backend_setup/tasks/get_vg_groupings.yml': line 3, column 3, but may\nbe elsewhere in the file depending on
the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Group devices by volume group
name, including existing devices\n  ^ here\n"}
```

**Comment by anoopcs9** [comment]
It seems like a [regression](https://github.com/ansible/ansible/commit/84e473a26ea95afec207409c42ba872c009fe6b6) with Ansible v2.12 was helping us hide this issue till its was [fixed](https://github.com/ansible/ansible/commit/3779c1f278685c5a8d7f78942ce649f6805a5775) with v2.13. Quoting some points from various related issues from Ansible:

* to_json and to_nice_json ALWAYS are supposed to return a string (of serialized JSON) otherwise they would be a Python dictionary/list. Ansible tends to automatically convert JSON into Python data structures, though we have some safeguards against it when explicitly using those filters (which would create the behavior you see).
* You should really only use to_json/to_nice_json on things that are Ansible variables

Therefore we must refrain from using `to_json` filter on variables within Jinja2 template.


Issue references: https://github.com/ansible/ansible/issues/77004, https://github.com/ansible/ansible/issues/76443

**Comment by dkelson** [comment]
This makes it so oVirt 4.5.4 with ovirt-node-ng-installer-latest-el8.iso (currently links to 4.5.4-2022120615) won't install.

**Comment by Dave3o3** [comment]
Until #136 is merged:

```bash
sed -i 's/output | to_json/output/' /etc/ansible/roles/gluster.infra/roles/backend_setup/tasks/get_vg_groupings.yml
sed -i 's/output | to_json/output/' /etc/ansible/roles/gluster.infra/roles/backend_setup/tasks/thick_lv_create.yml
sed -i 's/output | to_json/output/' /etc/ansible/roles/gluster.infra/roles/backend_setup/tasks/thin_pool_create.yml
sed -i 's/output | to_json/output/' /etc/ansible/roles/gluster.infra/roles/backend_setup/tasks/thin_volume_create.yml
```

**Comment by anoopcs9** [comment]
Fixed by #136
